### PR TITLE
Create constants for precompiles' 

### LINKED ISSUE #390: Create constants for precompiles' calldata lengths

### PR DESCRIPTION
**Pull request by iinaki**
closes #390

### CODE REVIEW

**Comment by juanimedone** [comment]
We have a `constants.rs` file so you could place the new constants there.

**Comment by iinaki** [comment]
> We have a `constants.rs` file so you could place the new constants there.

Done!

File: src/precompiles.rs [hunk 1]
```diff
@@ -260,33 +264,37 @@ pub fn ecpairing(
...
+                ..start + (G1_POINT_SIZE + ECP_FIELD_SIZE * 2)], // calldata[start + 96..start + 128]
+            &calldata[start + G1_POINT_SIZE..start + (G1_POINT_SIZE + ECP_FIELD_SIZE)], // calldata[start + 64..start + 96]
         ]
         .concat();
         let g2_y_bytes = [
-            &calldata[start + 160..start + 192],
-            &calldata[start + 128..start + 160],
+            &calldata[start + (G2_POINT_SIZE * ECP_FIELD_SIZE)
+                ..start + (G2_POINT_SIZE + ECP_FIELD_SIZE * 2)], // calldata[start + 160..start + 192]
+            &calldata[start + G2_POINT_SIZE..start + (G2_POINT_SIZE * ECP_FIELD_SIZE)], // calldata[start + 128..start + 160]
```
Comment (juanimedone) [inline src/precompiles.rs:297]: This fixes `ecpairing` tests:
```suggestion
            &calldata[start + (G2_POINT_SIZE + ECP_FIELD_SIZE)
                ..start + (G2_POINT_SIZE + ECP_FIELD_SIZE * 2)], // calldata[start + 160..start + 192]
            &calldata[start + G2_POINT_SIZE..start + (G2_POINT_SIZE + ECP_FIELD_SIZE)], // calldata[start + 128..start + 160]
```

File: src/precompiles.rs [hunk 2]
```diff
@@ -452,30 +460,30 @@ pub fn blake2f(
...
 
     t[0] = u64::from_le_bytes(
-        calldata[196..204]
+        calldata[BF2_OFFSET_COUNT_INIT..BF2_OFFSET_COUNT_INIT + BF2_VEC_ELEM_SIZE]
             .try_into()
             .map_err(|_| PrecompileError::InvalidCalldata)?,
     );
     t[1] = u64::from_le_bytes(
-        calldata[204..212]
+        calldata[BF2_OFFSET_COUNT_INIT + BF2_VEC_ELEM_SIZE * 2..BF2_BLOCK_FLAG]
```
Comment (juanimedone) [inline src/precompiles.rs:486]: This should fix `blake2f` tests:
```suggestion
        calldata[BF2_OFFSET_COUNT_INIT + BF2_VEC_ELEM_SIZE..BF2_BLOCK_FLAG]
```

**Review by juanimedone — CHANGES_REQUESTED** [review]

**Comment by juanimedone** [comment]
That should make all tests run successfully. In order to fix the Lint CI you can merge `fix-gas-consumption` (I've already fixed it there) into this branch.

File: src/precompiles.rs (hunk 1 above)
Comment (iinaki) [reply src/precompiles.rs]: Done!

File: src/precompiles.rs (hunk 2 above)
Comment (iinaki) [reply src/precompiles.rs]: Done!

**Review by juanimedone — APPROVED** [review]
LGTM

**Comment by juanimedone** [comment]
Maybe you can change the title of the PR before merging as the new added constants are not only calldata lengths.

**Review by igaray — APPROVED** [review]

**Review by MegaRedHand — APPROVED** [review]

**Pull request merged**